src/test/java/com/getindata/connectors/http/internal/table/lookup/HttpLookupTableSourceITCaseTest.java

```diff
@@ -169,9 +180,340 @@ public void testHttpsMTlsLookupJoin() throws Exception {
+            wireMockServer,
+            List.of(
+                matchingJsonPath("$.enrichedString"),
+                matchingJsonPath("$.row.aStringColumn"),
+                matchingJsonPath("$.row.anIntColumn"),
+                matchingJsonPath("$.row.aFloatColumn")
+            )
+        );
+
+        String fields =
+            "`row` ROW<`aStringColumn` STRING, `anIntColumn` INT, `aFloatColumn` FLOAT>\n";
+
+        String sourceTable =
+            "CREATE TABLE Orders (\n"
+                + "  proc_time AS PROCTIME(),\n"
+                + "  id STRING,\n"
+                + fields
+                + ") WITH ("
+                + "'connector' = 'datagen',"
+                + "'rows-per-second' = '1',"
+                + "'fields.id.kind' = 'sequence',"
+                + "'fields.id.start' = '1',"
+                + "'fields.id.end' = '5'"
+                + ")";
+
+        String lookupTable =
+            "CREATE TABLE Customers (\n" +
+                "  `enrichedInt` INT,\n" +
+                "  `enrichedString` STRING,\n" +
+                "  \n"
+                + fields
+                + ") WITH ("
+                + "'format' = 'json',"
+                + "'lookup-request.format' = 'json',"
+                + "'lookup-request.format.json.fail-on-missing-field' = 'true',"
+                + "'connector' = 'rest-lookup',"
+                + "'lookup-method' = 'POST',"
+                + "'url' = 'http://localhost:9090/client',"
+                + "'gid.connector.http.source.lookup.header.Content-Type' = 'application/json',"
+                + "'asyncPolling' = 'true'"
+                + ")";
+
+        tEnv.executeSql(sourceTable);
+        tEnv.executeSql(lookupTable);
+
+        // WHEN
+        // SQL query that performs JOIN on both tables.
+        String joinQuery =
+            "SELECT o.id, o.`row`, c.enrichedInt, c.enrichedString FROM Orders AS o"
+                + " JOIN Customers FOR SYSTEM_TIME AS OF o.proc_time AS c"
+                + " ON (\n"
+                + "  o.id = c.enrichedString AND\n"
+                + "  o.`row` = c.`row`\n"
+                + ")";
+
+        TableResult result = tEnv.executeSql(joinQuery);
+        result.await(15, TimeUnit.SECONDS);
+
+        // THEN
+        SortedSet<Row> collectedRows = getCollectedRows(result);
+
+        // TODO add assert on values
+        assertThat(collectedRows.size()).isEqualTo(5);
+    }
+
+    @Test
+    public void testLookupJoinOnRowWithRowType() throws Exception {
+
+        // GIVEN
+        setUpServerBodyStub(
+            "POST",
+            wireMockServer,
+            List.of(
+                matchingJsonPath("$.nestedRow.aStringColumn"),
+                matchingJsonPath("$.nestedRow.anIntColumn"),
+                matchingJsonPath("$.nestedRow.aRow.anotherStringColumn"),
+                matchingJsonPath("$.nestedRow.aRow.anotherIntColumn")
+            )
+        );
+
+        String fields = "  `nestedRow` ROW<" +
```

Comment:
I would add new line after `=` to increase readability.

Reply:
done